server/src/main/java/org/kaazing/gateway/server/context/resolve/DefaultServiceContext.java

```diff
@@ -729,7 +729,28 @@ private void injectServiceOptions(String transportURI, Map<String, Object> optio
                                 getProperties().get("encryption.key.alias"));
                         options.put(format(optionPattern, SERVICE_DOMAIN),
                                 getProperties().get("service.domain"));
+                    }
+                }
 
+                // TCP
+                for (String optionPattern : asList("tcp.%s")) {
+                    // NO REALM_NAME as this will be an accept/connect option
+                    String tcpRealmOptionName = format(optionPattern, REALM_NAME);
+                    String tcpRealmName = (String) options.get(tcpRealmOptionName);
+                    if (tcpRealmName != null) {
+                        // check if it's the same as the configured realm
+                        if (!serviceRealmContext.getName().equals(tcpRealmName)) {
+                            logger.error("{} configuration error: {} needs to be set to the same value as the configured realm",
+                                    serviceName, tcpRealmOptionName);
+                            throw new IllegalArgumentException(
+                                    tcpRealmOptionName + " needs to be the same as the configured realm");
+                        }
+                        options.put(format(optionPattern, LOGIN_CONTEXT_FACTORY), serviceRealmContext.getLoginContextFactory());
+//                    initial support will just be authenticate or not
+//                    options.put(format(optionPattern, REQUIRED_ROLES),
+//                            getRequireRoles());
+//                    options.put(format(optionPattern, REALM_USER_PRINCIPAL_CLASSES.name()),
+//                            getUserPrincipalClasses(serviceRealmContext.getUserPrincipalClasses()));
```

Comment:
A service context holds just one realm context and that is resolved to the realm value set in the '<service><realm-name>' tag. That's why I mentioned earlier that both `tcp.realm` and `realm-name` need to be specified in order for the realm context to be added to the service. I can resolve the service realm context using the `tcp.realm` accept option value when the `realm-name` is not specified, but that would require adding `tcp.realm` to the community xsd also.